### DISCUSSION TITLE
A question about the reasons for importing async hooks

### DISCUSSION THREAD

**Discussion by beita1** [tinylibs/tinypool#36] [Q&A]
HI,  [Aslemammad](https://github.com/Aslemammad)
I'm here again to ask a question about async hooks, hah
I don't understand why tinypool uses async hooks style?  As a performance-related library, the async hooks style will reduce performance.  What kind of problems must be solved by using async hooks?

Best Regards,
Kris Xu

**Comment by Aslemammad** [comment]
Yep! it reduces, but I'd like to see a benchmark specific for tinypool to see how things go if we remove it! the purpose more of why people use async_hooks (debugging, ...)! that's what was mentioned in Piscina!
(though I've never used that case for tinypool)

**Reply by beita1** [reply]
Ok, buddy. Let me prepare a benchmark. And in fact,  I am also writing a thread pool library. Let us see the design difference and exchange learning them in the future.
But this may be a long story. Recently I have been working very hard and very tired. And I will go out on my honeymoon next week.  haha~
Keep in touch with each other.

**Reply by Aslemammad** [reply]
Wow, congrats on the honeymoon! i'm open to do more changes to tinypool (no heavy breaking changes though)
Could we chat in discord? My discord: Aslemammad#6670

**Reply by beita1** [reply]
Yeah, I had the same idea.

**Comment by beita1** [comment]
已收到，谢谢
